_posts/volume/2023-03-12-volume-groups.md

```diff
@@ -0,0 +1,332 @@
+```
+
+**Step 3**  
+> Note: Before deleting your volume group make sure you do not have any
+> volumes in the group and all group snapshots have been removed.
+
+Now we will delete the volume group, make sure to replace the `<group-id>`
+with the ID of the volume group you want to remove.
+```bash
+openstack volume group delete <group-id> --os-volume-api-version 3.51
+```
+
+> Note: If you still have volumes attached and want to delete them as well you
+can add the `--force` argument to the command. This will delete all volumes
+attached to the volume group together with the Volume Group.
+
+Your volume group will now be deleted.
+
+---
+
+## Creating a volume group snapshot
+Creating a snapshot for a volume group is a simple process and can be done
+using the OpenStack Dashboard or the OpenStack CLI. In this article we will go
+through both methods.
+
+ - [Create a volume group snapshot using the OpenStack Dashboard](
+    #create-a-volume-group-snapshot-using-the-openstack-dashboard)
+ - [Create a volume group snapshot using the OpenStack CLI](
+    #create-a-volume-group-snapshot-using-the-openstack-cli)
+
+### Create a volume group snapshot using the OpenStack Dashboard
+**Step 1**
+Log in to the OpenStack Dashboard.
+
+**Step 2**
+Navigate to the `VOLUMES` section and click on `Groups`.
+
+**Step 3**
+Click on the `CREATE SNAPSHOT` button behind the volume group you want to
+create a snapshot for.
+
+The snapshots will now be created. This process can take a while depending on
+the size of the volumes in the volume group.
+
+### Create a volume group snapshot using the OpenStack CLI
+**Step 1**
+First make sure you have setup the OpenStack CLI and that you are able to
+execute commands using the `openstack` command. For more information please
+refer to the
+[Using the OpenStack CLI article](
+    {{ '/articles/using-the-cli-linux' | relative_url }}
+).
+
+**Step 2**
+First we will list all volume groups to get the ID of the volume group you
+want to create a snapshot for.
+```bash
+openstack volume group list --os-volume-api-version 3.51
+```
+
+**Step 3**
+We will now proceed to create the volume group snapshot, make sure to replace
+the `<group-id>` with the ID of the volume group you want to create a snapshot
+and `<name>` with the name you want to give the snapshot.
+```bash
+openstack volume group snapshot create --os-volume-api-version 3.51 <group-id> --name <name> 
+```
+
+## Restoring a volume group snapshot
+Sadly at the time of writing this article 03-2024 the OpenStack does not
+provide a way to restore a volume group snapshot. The snapshots are only 
+restorable individually. We recommend using the OpenStack Dashboard to restore
+a volume group snapshot or using the OpenStack CLI to restore the snapshots
+individually.
+
+You can find the OpenStack Snapshot in the OpenStack Dashboard by going to the
+`VOLUMES` section and then clicking on `Snapshots`.
+
+<!-- TODO: Add a link to a snapshot restore article >
```

Comment:
Please remove the todo from the final article

Reply:
I fixed the comment so it does not show on the page.
I do think we should keep it there so we can find it in the future.

Reply:
works for me :)